include/readout/concepts/RequestHandlerConcept.hpp

```diff
@@ -62,22 +65,26 @@ class RequestHandlerConcept
   // Request Result
   struct RequestResult
   {
-    RequestResult(ResultCode rc, dfmessages::DataRequest dr, unsigned rdus = 0)
+    RequestResult(ResultCode rc, dfmessages::DataRequest dr)
       : result_code(rc)
       , data_request(dr)
-      , request_delay_us(rdus)
+      , fragment()
+    {}
+    RequestResult(ResultCode rc, dfmessages::DataRequest dr, dataformats::Fragment&& frag)
+      : result_code(rc)
+      , data_request(dr)
+      , fragment(std::move(frag))
     {}
     ResultCode result_code;
     dfmessages::DataRequest data_request;
-    unsigned request_delay_us;
+    std::unique_ptr<dataformats::Fragment> fragment;
   };
 
   // Bookkeeping of OOB requests
   std::map<dfmessages::DataRequest, int> m_request_counter;
 
-  virtual RequestResult cleanup_request(dfmessages::DataRequest /*dr*/, unsigned /*delay_us*/ = 0) = 0;
-  virtual RequestResult data_request(dfmessages::DataRequest /*dr*/, unsigned /*delay_us*/ = 0) = 0;
-  virtual void executor() = 0;
+  virtual void cleanup() = 0;
```

Comment:
If this is a thread's function, use `run_` prefix. This is true for any other occasion below.